yt_dlp/extractor/mixch.py

```diff
@@ -116,3 +117,88 @@ def _real_extract(self, url):
+                         'ext': 'mp4'}],
+            **traverse_obj(data, {
+                'title': ('movie', 'title', {str_or_none}),
+                'thumbnail': ('movie', 'thumbnailURL', {url_or_none}),
+                'uploader': ('ownerInfo', 'name', {str_or_none}),
+                'uploader_id': ('ownerInfo', 'id', {int_or_none}),
+                'channel_follower_count': ('ownerInfo', 'fan', {int_or_none}),
+                'view_count': ('ownerInfo', 'view', {int_or_none}),
+                'like_count': ('movie', 'favCount', {int_or_none}),
+                'comment_count': ('movie', 'commentCount', {int_or_none}),
+            }),
+            'uploader_url': 'https://mixch.tv/u/' + traverse_obj(data, ('ownerInfo', 'id', {str_or_none})),
+            'live_status': 'not_live',
+            '__post_extractor': self.extract_comments(video_id),
+        }
+
+    def _get_comments(self, video_id):
+        data = self._download_json(f'https://mixch.tv/api-web/movies/{video_id}/comments?', video_id,
+                                   note='Downloading comments', errnote='Failed to download comments')
+        comment_dl_times = 1
+        MAX_DL_TIMES = 10
+        has_next = True
+        next_cursor = ''
+
+        # Comments are organized in a json chain, connected with 'nextCursor' property.
+        # There are up to 20 comments in one json file.
+        while has_next:
+            yield from traverse_obj(data,
+                                    ('comments', lambda k, v: v['comment'], {
+                                        'author': ('user_name', {str_or_none}),
+                                        'author_id': ('user_id', {int_or_none}),
+                                        'id': ('id', {int_or_none}),
+                                        'text': ('comment', {str_or_none}),
+                                        'timestamp': ('created', {int_or_none}),
+                                    }))
+
+            has_next = traverse_obj(data, ('hasNext'), {bool_or_none})
+            next_cursor = traverse_obj(data, ('nextCursor'), {str_or_none})
+
+            if comment_dl_times == MAX_DL_TIMES:
+                msg = '{video_id}: Comment count is {comment_count}. Only take first 200 comments into json.'.format(
+                    video_id=video_id, comment_count=traverse_obj(data, ('commentsCount', {int})))
+                self.to_screen(msg)
+                has_next = False
+
+            if has_next:
+                data = self._download_json(f'https://mixch.tv/api-web/movies/{video_id}/comments?cursor={next_cursor}&limit=20',
+                                           (video_id, next_cursor),
+                                           note='Downloading comments', errnote='Failed to download comments')
+                # Limit comments download times to avoid server forbidding.
+                comment_dl_times += 1
```

Comment:
I don't think the extractor should have a hard limit on the number of comments that can be extracted. Maybe there can be an extractor arg, but generally, I would say the user is responsible for picking an appropriate sleep interval.
If forbidden messages are common, we could implement some retry logic.

I would also simplify this a bit:
```suggestion
        base_url = f'https://mixch.tv/api-web/movies/{video_id}/comments'
        has_next = True
        next_cursor = ''

        while has_next:
            data = self._download_json(
                base_url, video_id, note='Downloading comments', errnote='Failed to download comments',
                query={'cursor': next_cursor, 'limit': 20})

            yield from traverse_obj(data, ('comments', lambda _, v: v['comment'], {
                'author': ('user_name', {str}),
                'author_id': ('user_id', {int_or_none}),
                'author_thumbnail': ('profile_image_url', {url_or_none}),
                'id': ('id', {int_or_none}),
                'text': ('comment', {str_or_none}),
                'timestamp': ('created', {int_or_none}),
            }))

            has_next = traverse_obj(data, ('hasNext'), {bool_or_none})
            next_cursor = traverse_obj(data, ('nextCursor'), {str_or_none})
```

Reply:
> If forbidden messages are common, we could implement some retry logic.

Retry might be unacceptable. Because the server will ban all access form downloader's IP for a period of time -- about several minutes?

I add two extractor args to limit extracting of comments.